server/bleep/src/query/parser.rs

```diff
@@ -22,6 +22,27 @@ pub enum Target<'a> {
     Content(Literal<'a>),
 }
 
+#[derive(Default, Clone, Debug, PartialEq, Eq)]
+pub struct NLQuery<'a> {
+    pub repo: Option<Literal<'a>>,
+    pub lang: Option<Cow<'a, str>>,
+    pub target: Option<Literal<'a>>,
+}
+
+impl<'a> NLQuery<'a> {
+    pub fn repo(&self) -> Option<&Cow<'_, str>> {
+        self.repo.as_ref().and_then(|t| t.as_plain())
```

Comment:
if `repo` is a regex here, it is silently converted to a `None` instead of raising a parse error.

Reply:
Will this term just be ignored then? E.g. `How do we parse a query? repo:/bloop/` would just be executed as `How do we parse a query?`

Reply:
yes, this term will just be ignored altogether. i haven't figured out regex filters yet, unsure if its possible through qdrant directly either, what we might need is some kind of query-builder that translates an NLQuery down to a Qdrant query in a better way.